src/core/core_utils.js

```diff
@@ -243,9 +243,10 @@ function isBooleanArray(arr, len) {
  */
 function isNumberArray(arr, len) {
   return (
-    Array.isArray(arr) &&
-    (len === null || arr.length === len) &&
-    arr.every(x => typeof x === "number")
+    (ArrayBuffer.isView(arr) && (len === null || arr.length === len)) ||
```

Comment:
Note that this will also return true for `BigInt64Array`, even if its elements do not contain values of type `"number"`.

Reply:
It has been addressed in #18526.